src/algebraic_geometry/projective_spectrum/topology.lean

```diff
@@ -0,0 +1,140 @@
+
+## Implementation note
+The type of `vanishing_ideal` is an `ideal` but instead of `homogeneous_ideal`; this is to take 
+advantage of `ideal_gc` so that setting up a galois connection is easier.
+-/
+
+noncomputable theory
+open_locale direct_sum big_operators pointwise
+open direct_sum set_like
+
+variables {R A: Type*}
+variables [comm_semiring R] [comm_ring A] [algebra R A]
+variables (𝒜 : ℕ → submodule R A) [graded_algebra 𝒜]
+
+/--
+The projective spectrum of a graded commutative ring is the subtype of all homogenous ideals that
+are prime and not containing all the irrelevant ideal.
+-/
+@[nolint has_inhabited_instance]
+def projective_spectrum :=
+{I : homogeneous_ideal 𝒜 // I.1.is_prime ∧ ¬(homogeneous_ideal.irrelevant 𝒜 ≤ I)}
+
+namespace projective_spectrum
+
+
+variable {𝒜}
+/-- A method to view a point in the projective spectrum of a graded ring
+as a homogeneous ideal of that ring. -/
+abbreviation as_homogeneous_ideal (x : projective_spectrum 𝒜) : homogeneous_ideal 𝒜 := x.val
+
+lemma as_homogeneous_ideal.explicit (x : projective_spectrum 𝒜) :
+  x.as_homogeneous_ideal = x.1 := rfl
+
+instance is_prime (x : projective_spectrum 𝒜) :
+  x.as_homogeneous_ideal.1.is_prime := x.2.1
+
+@[ext] lemma ext {x y : projective_spectrum 𝒜} :
+  x = y ↔ x.as_homogeneous_ideal = y.as_homogeneous_ideal :=
+subtype.ext_iff_val
+
+variable (𝒜)
+/-- The zero locus of a set `s` of elements of a commutative ring `A`
+is the set of all relevant prime homogeneous ideals of the ring that contain the set `s`.
+
+An element `f` of `A` can be thought of as a dependent function on the projective spectrum of `𝒜`.
+At a point `x` (a prime homogeneous ideal)
+the function (i.e., element) `f` takes values in the quotient ring `A` modulo the prime ideal `x`.
+In this manner, `zero_locus s` is exactly the subset of `projective_spectrum 𝒜`
+where all "functions" in `s` vanish simultaneously.
+-/
+def zero_locus (s : set A) : set (projective_spectrum 𝒜) :=
+{x | s ⊆ x.as_homogeneous_ideal}
+
+@[simp] lemma mem_zero_locus (x : projective_spectrum 𝒜) (s : set A) :
+  x ∈ zero_locus 𝒜 s ↔ s ⊆ x.as_homogeneous_ideal := iff.rfl
+
+@[simp] lemma zero_locus_span (s : set A) :
+  zero_locus 𝒜 (ideal.span s) = zero_locus 𝒜 s :=
+by { ext x, exact (submodule.gi _ _).gc s x.as_homogeneous_ideal }
+
+variable {𝒜}
+/-- The vanishing ideal of a set `t` of points
+of the prime spectrum of a commutative ring `R`
+is the intersection of all the prime ideals in the set `t`.
+
+An element `f` of `A` can be thought of as a dependent function on the projective spectrum of `𝒜`.
+At a point `x` (a prime homogeneous ideal)
+the function (i.e., element) `f` takes values in the quotient ring `A` modulo the prime ideal `x`.
+In this manner, `vanishing_ideal t` is exactly the ideal of `A`
+consisting of all "functions" that vanish on all of `t`.
+-/
+def vanishing_ideal (t : set (projective_spectrum 𝒜)) : ideal A :=
+⨅ (x : projective_spectrum 𝒜) (h : x ∈ t), x.as_homogeneous_ideal.1
```

Comment:
Can you not take this intersection directly in `homogeneous_ideal 𝒜`? You endowed it with a lattice structure, right?

Reply:
The main reason that I didn't take intersection as a homogeneous ideal is because we want a Galois connection in ideals so that lemmas like `le_vanishing_ideal_zero_locus` would work.

Reply:
I have updated the doc string to include some `##Implementation notes` to show this.

Reply:
There is a missing set_like instance for homogeneous ideal. I will make a quick pr.